OpenTween.csproj を新しい形式にする

kzrnm opened the pull request:
動作には影響がないようにしているつもりです。

これまでとの違いとして、リリース時に AssemblyInfo.cs のバージョンを書き換えていたのを OpenTween.csproj のバージョンを書き換えるようにする必要があります。

kzrnm committed:
OpenTween.csprojを新しい形式に

kzrnm committed:
ビルドスクリプトを新しい形式のcsprojに対応

kzrnm committed:
Microsoft.NET.Sdk ではデフォルトで Deterministic=true

upsilon commented:
SDK スタイルの csproj は OpenTween.Tests.csproj では移行済みですが、OpenTween.csproj は Visual Studio 上で WinForms のフォームやユーザーコントロールの新規追加ができなくなる問題があり移行できていません。
.NET に移行できればこの辺の問題も解決できそうですが、 .NET 6 の時点では HiDPI 対応に難があり今のところ保留となっています…。

![image](https://user-images.githubusercontent.com/92577/188164315-655a6911-1739-41e8-83be-7f6d92a428ac.png)

kzrnm committed:
Add UseWindowsForms

kzrnm commented:
@upsilon

確認しました。


----
`<UseWindowsForms>true</UseWindowsForms>` を追加したら出そうなのですがどうでしょう？

.NET Framework で使う分にはビルドの影響もないはずです。

![image](https://user-images.githubusercontent.com/32071278/188175019-661ef4e3-f9e0-4791-8754-acf8ed8482cb.png)

![image](https://user-images.githubusercontent.com/32071278/188175349-c76414d0-fc1f-4a2a-b9ff-9ec7eb0edae9.png)

upsilon committed:
GenerateSatelliteAssembliesForCore を有効にする

従来はサテライトアセンブリを生成する際に deterministic オプションを有効にする
ために build-satelite-assembly.ps1 で独自に csc.exe を呼び出していたが、
csproj の設定で GenerateSatelliteAssembliesForCore を有効にすることで同等の
効果を得ることができたためこれに置き換えた

https://learn.microsoft.com/en-us/dotnet/core/extensions/create-satellite-assemblies#compile-satellite-assemblies

upsilon commented:
`AssemblyInfo.cs` の自動生成を有効にすると、ビルド時のロケール設定に依存して `OpenTween.pdb` および `OpenTween.exe` のハッシュ値が変化してしまい Reproducible build の妨げになることが分かったため、回避策が見つかるまでは `<GenerateAssemblyInfo>false</GenerateAssemblyInfo>` で自動生成を無効化します。

```diff
--- a/OpenTween/obj/Debug/net472/OpenTween.AssemblyInfo.cs        2022-11-29 18:28:10.000000000 +0900
+++ b/OpenTween/obj/Debug/net472/OpenTween.AssemblyInfo.cs        2022-12-01 00:24:09.904235300 +0900
@@ -1,10 +1,10 @@
 //------------------------------------------------------------------------------
 // <auto-generated>
-//     This code was generated by a tool.
-//     Runtime Version:4.0.30319.42000
+//     このコードはツールによって生成されました。
+//     ランタイム バージョン:4.0.30319.42000
 //
-//     Changes to this file may cause incorrect behavior and will be lost if
-//     the code is regenerated.
+//     このファイルへの変更は、以下の状況下で不正な動作の原因になったり、
+//     コードが再生成されるときに損失したりします。
 // </auto-generated>
 //------------------------------------------------------------------------------

@@ -23,5 +23,5 @@
 [assembly: System.Runtime.CompilerServices.InternalsVisibleTo("OpenTween.Tests")]
 [assembly: System.Runtime.CompilerServices.InternalsVisibleTo("DynamicProxyGenAssembly2")]

-// Generated by the MSBuild WriteCodeFragment class.
+// MSBuild WriteCodeFragment クラスによって生成されました。

```

upsilon committed:
AssemblyInfo.cs の自動生成を無効化

生成されるソースコードに含まれるコメントがロケール設定に依存して変化してしまい
Reproducible build の妨げとなる問題があるため、回避策が見つかるまで無効化する。
